src/oscar/apps/order/abstract_models.py

```diff
@@ -1169,3 +1177,27 @@ def description(self):
         if self.voucher_code:
             return self.voucher_code
         return self.offer_name or ""
+
+
+class AbstractSurcharge(models.Model):
+    order = models.ForeignKey(
+        'order.Order',
+        on_delete=models.CASCADE,
+        related_name="surcharges",
+        verbose_name=_("Surcharges"))
+
+    name = models.CharField(
```

Comment:
Suggest we add a `code` field here, which is the non-translatable identifier for a charge.